File: LAB3/JAVA/LAB3POP/src/Consumer.java
public class Consumer implements Runnable{
    private final int _maxItem;
    private final Storage _storage;

    public Consumer(int maxItem, Storage storage) {
        _maxItem = maxItem;
        _storage = storage;
    }

    @Override
    public void run() {
        for (int i = 0; i < _maxItem; i++) {
            String item;
            try {
                _storage.empty.acquire();
                Thread.sleep(1000);
                _storage.access.acquire();

                item = _storage.getItem();
                _storage.removeItem();
                System.out.println("Took " + item);

                _storage.access.release();
                _storage.full.release();

            } catch (InterruptedException e) {
                e.printStackTrace();
            }
        }
    }
}

File: LAB3/C#/Lab3POP/Lab3POP/Consumer.cs
using System;
using System.Collections.Generic;
using System.Data.SqlTypes;
using System.Linq;
using System.Text;
using System.Threading;
using System.Threading.Tasks;

namespace Lab3POP
{
    internal class Consumer
    {
        private int _maxItem = 0;
        private Storage _storage;
        public Consumer(int maxItem, Storage storage)
        {
            _maxItem = maxItem;
            _storage = storage;
        }

        public void TakeItem()
        {
            for (int i = 0; i < _maxItem; i++)
            {
                _storage.AcquireEmpty();
                Thread.Sleep(1000);
                _storage.AcquireAccess();

                string item = _storage.GetItem();
                _storage.RemoveItem();

                _storage.ReleaseFull();

                _storage.ReleaseAccess();

                Console.WriteLine("Took " + item);
            }
        }
    }
}
